astropy/io/ascii/tdat.py

```diff
@@ -0,0 +1,815 @@
+        # FloatType) for each col.
+        self._convert_vals(cols)
+
+        t_cols = [
+            np.ma.MaskedArray(x.data, mask=x.mask)
+            if hasattr(x, "mask") and np.any(x.mask)
+            else x.data
+            for x in cols
+        ]
+        out = core.Table(t_cols, names=[x.name for x in cols], meta=meta["table"])
+        indices = []
+        for col, out_col in zip(cols, out.columns.values()):
+            for attr in ("format", "unit", "description"):
+                if hasattr(col, attr):
+                    setattr(out_col, attr, getattr(col, attr))
+            if hasattr(col, "meta"):
+                out_col.meta.update(col.meta)
+                if "index" in col.meta:
+                    if col.meta["index"] == "key":
+                        indices.insert(0, col.name)
+                    else:
+                        indices.append(col.name)
+        # Add indices, if specified
+        if len(indices) > 0:
+            for name in indices:
+                out.add_index(name)
+        return out
+
+
+class Tdat(core.BaseReader):
+    """TDAT format
+    See: https://heasarc.gsfc.nasa.gov/docs/software/dbdocs/tdat.html"
+
+    Example::
+      <HEADER>
+      # # and // are comments
+      table_name = example_table
+      table_description = "Example table"
+      #
+      # Table Parameters
+      #
+      field[id] = integer [meta.id] (key) // Unique ID
+      field[ra] = float:.4f_degree [pos.eq.ra] (index) // Right Ascension
+      field[name] = char12 [meta.id] // Name
+      #
+      # Virtual Parameters
+      #
+      table_author = Example et al.
+      #
+      # Data Format Specification
+      #
+      line[1] = id name ra
+      <DATA>
+      1|TargetOne|1.0|
+      2|TargetTwo|2.0|
+      <END>
+
+    The comments and keywords defined in the header, excepting common header
+    section titles and blank comments, are available via the output table
+    ``meta`` attribute::
+
+      >>> from astropy.io import ascii
+      >>> lines = ascii.tdat.make_example_data()
+      >>> data = ascii.read(lines, format='tdat')
+      >>> print(data.meta['comments'])
+      ['# and // are comments']
+      >>> for name, keyword in data.meta['keywords'].items():
+      ...   print(name, keyword)
+      table_name example_table
+      table_description Example table
+      table_author Example et al.
+
+
+    When writing to the TDAT format, the header will be auto-populated by
+    information in the Table, prioritizing information given in the Table.meta:
+
+    comments : list or string, (optional)
+    Table information which provide context. This information is
+    included in the header preceding all other lines and commented
+    out with #
+
+    keywords : dict, (optional, recommended)
+    Header keywords which will appear in the file as "name=value" lines.
+    Of particular importance are table_name, table_description,
+    and table_document_url.
```

Comment:
You can format this more nicely like below, so it looks like https://docs.astropy.org/en/stable/io/ascii/read.html#parameters-for-read
```suggestion
    **comments** : list or string, (optional)
      Table information which provide context. This information is
      included in the header preceding all other lines and commented
      out with #

    **keywords** : dict, (optional, recommended)
      Header keywords which will appear in the file as "name=value" lines.
      Of particular importance are table_name, table_description,
      and table_document_url.
```